### DISCUSSION TITLE
Installing on a "NO RPMB" Nexus 5X

### DISCUSSION THREAD

**Discussion by Tachi107** [Divested-Mobile/DivestOS-Build#60] [Q&A] [answered]
Hi, I'd really like to try DivestOS on my Nexus 5X (bullhead), but I'm a bit scared of bricking it. Let me explain.

"NO RPMB" basically means that you cannot permanently unlock your bootloader, as your motherboard lacks some firmware/chip that keeps track of its state. You can find more details on [this Reddit thread](https://www.reddit.com/r/nexus5x/comments/5hj0r0/bootloop_after_lg_replaced_mainboard_bootloader/).

Being unable to unlock the bootloader is really limiting and can make flashing of custom roms _almost_ impossible. But as you can see in the thread I linked it is possible to do it.

Now, that procedure isn't really detailed and the guy who described it also installs Magisk, su addons, TWRP, etc, while I'd like to install DivestOS as cleanly as possible (no root/Magisk, no custom recovery if possible, working Verified Boot 1.0).

I didn't find any information about how to make Verified Boot 1.0 work on the DivestOS website, and as I'm in a not so common situation I'd like to ask here before breaking stuff. Also, I installed LineageOS on my phone a few years ago, and I forgot how I worked around the locked bootloader :/

**Comment by SkewedZeppelin** [comment, marked as answer]
For verified boot 1.0 you simply install it, once locked it'll be enforcing.
There is no additional key you need to flash like 2.0.

As to the specifics of this bullhead issue I have no idea.
I know LG did a ton of silly things with that device.
In addition to the lovely bootloop issue for which you must have one of the
lucky few working devices left.

If you are able to activate OEM unlocking in developer settings, reboot to
fastboot, fastboot oem unlock, fastboot flash the DivestOS recovery, and then
adb sideload the full DivestOS image you should be good to go.

The thread seems to imply those steps should work, but you'll be locked by
default.

bullhead is largely tested working.
The OEM unlocking flag should persist unless you perform a factory reset via the
Settings app in DivestOS.
So even if DivestOS fails to boot, you should still be able to unlock again and
reflash stock or another ROM.

In this special case I definetly would expect the possibility of brick and plan
for it.
On a normal N5X at least you can recover from such brick via LAF/LG UP.

Any other questions feel free to ask.
Best of luck!

**Reply by Tachi107** [reply]
Alright, I flashed everything. It took a me a bit to remember how this stuff works, but I think that I managed to not brick it. I still don't know though, as I'm waiting for it to complete the first boot (that started 20 minutes ago).
Thank you for the reply, I thought I had to manually enroll some keys :)

**Reply by Tachi107** [reply]
I booted the phone one hour ago and it is still showing the boot animation :/
Edit: re-flashed everything from scratch and it worked flawlessly! Thanks again :D

**Comment by SkewedZeppelin** [comment]
No device takes longer than 10 minutes to boot.
Reboot to the recovery and do a factory reset.

**Reply by Tachi107** [reply]
See my updated comment (https://github.com/Divested-Mobile/DivestOS-Build/discussions/60#discussioncomment-2258378). I'm now using DivestOS and so far it seems really nice :)
